osu.Game/Screens/Multi/RealtimeMultiplayer/Participants/ParticipantPanel.cs

```diff
@@ -0,0 +1,187 @@
+                            },
+                            new FillFlowContainer
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Spacing = new Vector2(10),
+                                Direction = FillDirection.Horizontal,
+                                Children = new Drawable[]
+                                {
+                                    new UpdateableAvatar
+                                    {
+                                        Anchor = Anchor.CentreLeft,
+                                        Origin = Anchor.CentreLeft,
+                                        RelativeSizeAxes = Axes.Both,
+                                        FillMode = FillMode.Fit,
+                                        User = User.User
+                                    },
+                                    new UpdateableFlag
+                                    {
+                                        Anchor = Anchor.CentreLeft,
+                                        Origin = Anchor.CentreLeft,
+                                        Size = new Vector2(30, 20),
+                                        Country = User.User.Country
+                                    },
+                                    new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.CentreLeft,
+                                        Origin = Anchor.CentreLeft,
+                                        Font = OsuFont.GetFont(weight: FontWeight.Bold, size: 18),
+                                        Text = User.User.Username
+                                    },
+                                    new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.CentreLeft,
+                                        Origin = Anchor.CentreLeft,
+                                        Font = OsuFont.GetFont(size: 14),
+                                        Text = User.User.CurrentModeRank != null ? $"#{User.User.CurrentModeRank}" : string.Empty
+                                    }
+                                }
+                            },
+                            readyMark = new ReadyMark
+                            {
+                                Anchor = Anchor.CentreRight,
+                                Origin = Anchor.CentreRight,
+                                Margin = new MarginPadding { Right = 10 },
+                                Alpha = 0
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        protected override void OnRoomChanged()
+        {
+            base.OnRoomChanged();
+
+            if (Room == null)
+                return;
+
+            if (User.State == MultiplayerUserState.Ready)
+                readyMark.FadeIn(50);
```

Comment:
nit: would extract this transition duration to a const

Reply:
Sure. I've made it a const local to this method. I'm not going to move it outside because this is also super temporary - I'd implement this differently like giving the ready mark a y-movement, but I'm not spending too much time on designs at this point.